nodelet/src/callback_queue_manager.cpp

```diff
@@ -123,7 +123,8 @@ void CallbackQueueManager::removeQueue(const CallbackQueuePtr& queue)
   boost::mutex::scoped_lock lock(queues_mutex_);
   ROS_ASSERT(queues_.find(queue.get()) != queues_.end());
```

Comment:
This `ROS_ASSERT` has the same condition you're testing in the new `if` below. Not a big deal, but I'd suggest removing the `ROS_ASSERT` in the same commit that adds the `if`:

```suggestion
```

Strictly speaking, `removeQueue` should never get called with a `queue` that isn't in the `queues_`. Do you know why that could have happened?

Reply:
No, I have not been able to figure out why it is being called with a queue not in the queues_ list. It does not make sense because the argument passed in should never have been removed. https://github.com/clearpathrobotics/nodelet_core/blob/indigo-devel/nodelet/src/loader.cpp#L197

Reply:
Do you have a coredump showing the addresses of `st_queue` and `mt_queue`?

https://github.com/clearpathrobotics/nodelet_core/blob/a7da9266c766f01cfe667363f00104d7cfcb5f68/nodelet/src/loader.cpp#L184-L185

I wonder if both are set, and different. The only case I can think of that could lead to `queues_` not having one of them is if both `st_queue` and `mt_queue` are `NULL`.

Reply:
I looked at a core dump related to this and `st_queue != mt_queue != NULL`. The crash also happens in the first call to `removeQueue`, and it comes from the loader running in https://github.com/clearpathrobotics/nodelet_core/blob/indigo-devel/nodelet/src/loader.cpp#L294, which suggests the pluginlib subsystem was having trouble to find the nodelet plugin name. Maybe there's a corner case when that code runs, which leads to this.

Either way, I'm approving this for now as is, since I don't have a better solution atm.

Let me know if you don't have merge permissions, and I'd merge this.